api/src/main/java/io/opencensus/common/Duration.java

```diff
@@ -88,5 +88,21 @@ public static Duration fromMillis(long millis) {
    */
   public abstract int getNanos();
 
+  /**
+   * Compares the length of this {@code Duration} to the specified {@code Duration}.
+   *
+   * @param otherDuration the other {@code Duration} to compare to, not {@code null}.
+   * @return the comparator value: zero if equal, negative if this duration is shorter
+   *     than otherDuration, positive if larger.
+   * @throws NullPointerException if otherDuration is {@code null}.
+   */
+  public int compareLength(Duration otherDuration) {
+    int cmp = compareLong(Math.abs(getSeconds()), Math.abs(otherDuration.getSeconds()));
```

Comment:
I think that a positive length should be greater than a negative length, similar to a unit-less comparison.

Reply:
From my understanding lengths are all positive and should be the absolute value of Duration, while Duration could have negative values. I feel it's a little counterintuitive to have negative length.